### PR TITLE
Initial commit for Jenkins README

### LINKED ISSUE #123: Add README.md to buildenv/jenkins
At minimum, we should have a readme in the jenkins folder which documents what builds we have and how they run. If it becomes too large we can consider adding a doc folder there too.

related: #91 #122 #175 #176 eclipse/openj9-omr#3  eclipse/openj9-omr#4

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Provide list of builds with
    - Build name
    - Jenkins build badge
    - Description
    - Trigger
- Provide instructions for launching pull request builds
- Best practices when adding new builds
- link to Jenkins instance

Fixes #123

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

File: buildenv/jenkins/README.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        - Poll Github repo for changes
+
+- Promote-OpenJ9-OMR-master-to-openj9
+    - [![Build Status](https://ci.eclipse.org/openj9/buildStatus/icon?job=Promote-OpenJ9-OMR-master-to-openj9)](https://ci.eclipse.org/openj9/job/Promote-OpenJ9-OMR-master-to-openj9)
+    - Description:
+        - Promotes eclipse/openj9-omr branch master to branch openj9
+    - Trigger:
+        - Last step of `Pipeline-OMR-Acceptance`
+ 
+#### Pipleines
```
Comment (smlambert) [inline buildenv/jenkins/README.md]: typo

File: buildenv/jenkins/README.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    - Description:
+        - Compile and Sanity tests on linux_ppc-64_cmprssptrs_le
+    - Trigger:
+        - Github PR comment `Jenkins test sanity`
+
+#### Test
+
+- Test-Extended-linux_390-64_cmprssptrs
+    - [![Build Status](https://ci.eclipse.org/openj9/buildStatus/icon?job=Test-Extended-linux_390-64_cmprssptrs)](https://ci.eclipse.org/openj9/job/Test-Extended-linux_390-64_cmprssptrs)
+    - Descirption:
```
Comment (smlambert) [inline buildenv/jenkins/README.md]: typo

File: buildenv/jenkins/README.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+### Overview of Builds
+
+#### Build
+
+- Build-linux_390-64_cmprssptrs
+    - [![Build Status](https://ci.eclipse.org/openj9/buildStatus/icon?job=Build-linux_390-64_cmprssptrs)](https://ci.eclipse.org/openj9/job/Build-linux_390-64_cmprssptrs)
+    - Description:
+        - Compiles on linux_390-64_cmprssptrs
+        - Archives the SDK and test material
```
Comment (smlambert) [inline buildenv/jenkins/README.md]: If this is only a build job, probably no test material archived.

File: buildenv/jenkins/README.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    - Trigger:
+        - This job is used in other pipelines but can be launched manually
+
+- Build-linux_ppc-64_cmprssptrs_le
+    - [![Build Status](https://ci.eclipse.org/openj9/buildStatus/icon?job=Build-linux_ppc-64_cmprssptrs_le)](https://ci.eclipse.org/openj9/job/Build-linux_ppc-64_cmprssptrs_le)
+    - Description
+        - Compiles on linux_ppc-64_cmprssptrs_le
+        - Archives the SDK and test material
```
Comment (smlambert) [inline buildenv/jenkins/README.md]: same comment as line 44

File: buildenv/jenkins/README.md (hunk 3 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/README.md]: The purpose here for archiving the test source is to pass both the sdk and test material to the Test Jobs. That way we don't need to reclone it in the test job and we ensure we get the same source version as the build. We could not archive it and pass a SHA, but as I said, we would have to clone again. Not sure if one method is better than the other though.

**Comment by AdamBrousseau** [comment]
Any other comments/concerns or can this be merged?
@smlambert @DanHeidinga

File: buildenv/jenkins/README.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+- Current available platforms are Linux s390x and Linux PPCLE
+- OpenJ9 committers can request builds by commenting in a pull request
+- Request a build in a PR by commenting
+    - `Jenkins compile`
+    - `Jenkins test sanity`
+    - `Jenkins test extended`
+- Request a single platform instead of "all"
+    - `Jenkins test sanity zlinux`
+    - `Jenkins test sanity plinux`
+- You can also request a "Compile & Sanity" build from an openj9-openjdk-jdk9 PR or an openj9-omr PR.
```
Comment (DanHeidinga) [inline buildenv/jenkins/README.md]: Can make the extensions repo and openj9-omr repo mentions into links?

File: buildenv/jenkins/README.md (hunk 3 above)
Comment (DanHeidinga) [reply buildenv/jenkins/README.md]: Can you change the wording then to something like "Archives the SDK and test material for later pipeline stages"?

File: buildenv/jenkins/README.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        - Archives the SDK and test material
+    - Trigger:
+        - This job is used in other pipelines but can be launched manually
+
+#### Infrastructure
+
+- Mirror-OMR-to-OpenJ9-OMR
+    - [![Build Status](https://ci.eclipse.org/openj9/buildStatus/icon?job=Mirror-OMR-to-OpenJ9-OMR)](https://ci.eclipse.org/openj9/job/Mirror-OMR-to-OpenJ9-OMR)
+    - Description:
+        - Mirrors eclipse/omr/master to eclipse/openj9-omr/master
```
Comment (DanHeidinga) [inline buildenv/jenkins/README.md]: links would be great for the repos here.

File: buildenv/jenkins/README.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+        - Build periodically, 15 minutes
+
+- Mirror-OpenJ9-Website-to-Eclipse
+    - [![Build Status](https://ci.eclipse.org/openj9/buildStatus/icon?job=Mirror-OpenJ9-Website-to-Eclipse)](https://ci.eclipse.org/openj9/job/Mirror-OpenJ9-Website-to-Eclipse)
+    - Description:
+        - Mirrors github.com/eclipse/openj9-website to the Eclipse.org repo
+    - Trigger:
+        - Poll Github repo for changes
+
+- Promote-OpenJ9-OMR-master-to-openj9
```
Comment (DanHeidinga) [inline buildenv/jenkins/README.md]: Can we reorganize this a little so the two OMR related builds are together and the website one comes after?

Also, can this describe the tag laid down on merges?

**Comment by AdamBrousseau** [comment]
Pushed a new commit with, I believe, all the requested changes.

**Comment by AdamBrousseau** [comment]
pushed new commit with copyright header

**Pull request merged**